File: Prob#3 - Numbericalization/digital_number.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

typedef vector<char> vc;

class DigitalNumber {
   public:
    DigitalNumber(string number, int ver_scale, int hor_scale);
    void set_number(string number);
    void set_ver_scale(int ver_scale);
    void set_hor_scale(int hor_scale);
    void print_canvas();

   private:
    string number;
    vector<vc> canvas;
    int ver_scale, hor_scale;
};

DigitalNumber::DigitalNumber(string number, int ver_scale, int hor_scal) {
    // initialize canvas (set all place with NULL)
    int number_len = number.length();
    int canvas_ver_len = 3 * ver_scale;
    int canvas_hor_len = 3 * hor_scale * number_len;
    for (int i = 0; i < canvas_ver_len; i++) {
        vc canvas_hor(canvas_hor_len, NULL);
        canvas.push_back(canvas_hor);
    }

    // set number, ver_scale and hor_scale with user input
    set_number(number);
    set_ver_scale(ver_scale);
    set_hor_scale(hor_scale);
}

void DigitalNumber::set_number(string number) {
    this->number = number;
}

void DigitalNumber::set_ver_scale(int ver_scale) {
    this->ver_scale = ver_scale;
}

void DigitalNumber::set_hor_scale(int hor_scale) {
    this->hor_scale = hor_scale;
}

void DigitalNumber::print_canvas() {
    for (vc canvas_hor : canvas) {
        for (char c : canvas_hor) {
            cout << c;
        }
        cout << endl;
    }
}

File: Prob#3 - Numbericalization/src/DigitalNumber.java
public class DigitalNumber {
    public static int defaultScale = 5;
    private String number;
    private int verScale;
    private int horScale;
    private char canvas[][];

    DigitalNumber(String number, int verScale, int horScale) {
        setNumber(number);
        setHorScale(horScale);
        setVerScale(verScale);

        int numberLength = number.length();
        int verticalLength = defaultScale * verScale;
        int horizontalLength = defaultScale * horScale * numberLength + (numberLength * horScale);
        canvas = new char[verticalLength][horizontalLength];
    }

    void drawCanvas() {
        for (int i = 0; i < canvas.length; i++) {
            for (int j = 0; i < canvas[i].length; j++) {
                int currentNumber = Character.getNumericValue(number.charAt(j % (defaultScale * horScale + horScale)));
                int posI = i / verScale;
                int posJ = j / horScale;
                canvas[i][j] = posJ >= 5 ? ' ' : NumberTemplate.NUMBER_TEMPLATE[currentNumber][posI][posJ];
            }
        }
    }

    void printCanvas() {
        for (char[] ver : canvas) {
            for (char c : ver) {
                System.out.println(c);
            }
            System.out.println();
        }
    }

    public void setNumber(String number) {
        this.number = number;
    }

    public void setVerScale(int verScale) {
        this.verScale = verScale;
    }

    public void setHorScale(int horScale) {
        this.horScale = horScale;
    }
}
